tests/linalg_test.py

```diff
@@ -1620,6 +1620,55 @@ def testToeplitzConstructionWithKnownCases(self):
       [3, 2, 1, 4]], dtype=np.float32))
 
 
+  @jtu.sample_product(
+    shape=[[2, 1], [1, 2], [2, 5], [5, 2], [4, 20], [20, 4]],
+    dtype=float_types + complex_types)
+  def testNullSpace(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    args_maker = lambda: [X]
+    self._CheckAgainstNumpy(jtu.promote_like_jnp(osp.linalg.null_space),
+                            jsp.linalg.null_space, args_maker)
+
+  @jtu.sample_product(
+    shape=[[2, 1], [1, 2], [2, 5], [5, 2], [4, 20], [20, 4]],
+    dtype=float_types + complex_types,
+    size=[None, 2, 9])
+  def testNullSpaceCompiled(self, shape, dtype, size):
+    # test compiled version
+    rng = jtu.rand_default(self.rng())
+    X = rng(shape, dtype)
+    if size != None:
```

Comment:
For comparisons with `None`, we should use object identity:
```
if size is not None
```

Reply:
ofc